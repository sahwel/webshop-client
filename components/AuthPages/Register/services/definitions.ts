export enum Section {
  firstSection = 1,
  secondSection,
  thirdSection,
}
