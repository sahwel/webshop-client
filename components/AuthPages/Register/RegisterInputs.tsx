import dynamic from 'next/dynamic';
import React, { FunctionComponent } from 'react';
import InputContainer from '../../common/Form/InputContainer';
import FirstSection from './Sectionts/FirstSection';
import { Section } from './services/definitions';
const SecondSection = dynamic(() => import('./Sectionts/SecondSection'), {
  ssr: false,
});

interface RegisterInputsProps {
  section: Section;
}

const RegisterInputs: FunctionComponent<RegisterInputsProps> = ({
  section,
}) => {
  return (
    <InputContainer>
      {section === Section.firstSection && <FirstSection />}
      {section === Section.secondSection && <SecondSection />}
    </InputContainer>
  );
};

export default RegisterInputs;
