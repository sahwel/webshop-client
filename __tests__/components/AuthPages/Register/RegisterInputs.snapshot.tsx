import { render, renderHook } from '@testing-library/react';
import RegisterInputs from '@/components/AuthPages/Register/RegisterInputs';
import { Section } from '@/components/AuthPages/Register/services/definitions';

describe('RegisterInputs Snapshot with first section', () => {
  it('renders RegisterInputs with first sectionunchanged', () => {
    const { container } = render(
      <RegisterInputs section={Section.firstSection} />
    );
    expect(container).toMatchSnapshot();
  });
});

describe('RegisterInputs Snapshot with second section', () => {
  it('renders RegisterInputs with second sectionunchanged', () => {
    const { container } = render(
      <RegisterInputs section={Section.secondSection} />
    );
    expect(container).toMatchSnapshot();
  });
});

describe('RegisterInputs Snapshot with third section', () => {
  it('renders RegisterInputs with third sectionunchanged', () => {
    const { container } = render(
      <RegisterInputs section={Section.thirdSection} />
    );
    expect(container).toMatchSnapshot();
  });
});
