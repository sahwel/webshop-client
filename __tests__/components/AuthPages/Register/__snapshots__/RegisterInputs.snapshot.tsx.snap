// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterInputs Snapshot with first section renders RegisterInputs with first sectionunchanged 1`] = `
<div>
  <div
    class="input_container"
  >
    <p>
      Please enter some of your personal datas, then hit next!
    </p>
    <div
      class="input_container"
    >
      <input
        class="input"
        placeholder="Email"
        type="text"
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-envelope input_icon icon-7"
        data-icon="envelope"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 128C0 92.65 28.65 64 64 64H448C483.3 64 512 92.65 512 128V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V128zM48 128V150.1L220.5 291.7C241.1 308.7 270.9 308.7 291.5 291.7L464 150.1V127.1C464 119.2 456.8 111.1 448 111.1H64C55.16 111.1 48 119.2 48 127.1L48 128zM48 212.2V384C48 392.8 55.16 400 64 400H448C456.8 400 464 392.8 464 384V212.2L322 328.8C283.6 360.3 228.4 360.3 189.1 328.8L48 212.2z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="input_container"
    >
      <input
        class="input"
        placeholder="First name"
        type="text"
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-face-grin input_icon icon-7"
        data-icon="face-grin"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M349.5 308.4C368.2 303.1 385.4 320.4 374.1 336.5C350.4 374.6 306.3 399.1 255.9 399.1C205.6 399.1 161.5 374.6 136.9 336.5C126.5 320.4 143.7 303.1 162.3 308.4C191.3 315.1 222.8 318.8 255.9 318.8C289 318.8 320.6 315.1 349.5 308.4zM208.4 208C208.4 225.7 194 240 176.4 240C158.7 240 144.4 225.7 144.4 208C144.4 190.3 158.7 176 176.4 176C194 176 208.4 190.3 208.4 208zM304.4 208C304.4 190.3 318.7 176 336.4 176C354 176 368.4 190.3 368.4 208C368.4 225.7 354 240 336.4 240C318.7 240 304.4 225.7 304.4 208zM512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256zM256 48C141.1 48 48 141.1 48 256C48 370.9 141.1 464 256 464C370.9 464 464 370.9 464 256C464 141.1 370.9 48 256 48z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="input_container"
    >
      <input
        class="input"
        placeholder="Last name"
        type="text"
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-face-grin input_icon icon-7"
        data-icon="face-grin"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M349.5 308.4C368.2 303.1 385.4 320.4 374.1 336.5C350.4 374.6 306.3 399.1 255.9 399.1C205.6 399.1 161.5 374.6 136.9 336.5C126.5 320.4 143.7 303.1 162.3 308.4C191.3 315.1 222.8 318.8 255.9 318.8C289 318.8 320.6 315.1 349.5 308.4zM208.4 208C208.4 225.7 194 240 176.4 240C158.7 240 144.4 225.7 144.4 208C144.4 190.3 158.7 176 176.4 176C194 176 208.4 190.3 208.4 208zM304.4 208C304.4 190.3 318.7 176 336.4 176C354 176 368.4 190.3 368.4 208C368.4 225.7 354 240 336.4 240C318.7 240 304.4 225.7 304.4 208zM512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256zM256 48C141.1 48 48 141.1 48 256C48 370.9 141.1 464 256 464C370.9 464 464 370.9 464 256C464 141.1 370.9 48 256 48z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`RegisterInputs Snapshot with second section renders RegisterInputs with second sectionunchanged 1`] = `
<div>
  <div
    class="input_container"
  >
    <div
      class="container"
    >
      <svg
        class="loading"
        fill="none"
        role="status"
        viewBox="0 0 100 101"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
          fill="currentColor"
        />
        <path
          d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
          fill="currentFill"
        />
      </svg>
      <p
        class="text-2xl"
      >
        Loading...
      </p>
    </div>
  </div>
</div>
`;

exports[`RegisterInputs Snapshot with third section renders RegisterInputs with third sectionunchanged 1`] = `
<div>
  <div
    class="input_container"
  >
    <div
      class="container"
    >
      <svg
        class="loading"
        fill="none"
        role="status"
        viewBox="0 0 100 101"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
          fill="currentColor"
        />
        <path
          d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
          fill="currentFill"
        />
      </svg>
      <p
        class="text-2xl"
      >
        Loading...
      </p>
    </div>
  </div>
</div>
`;
